--- netsurf-3.3/monkey/Makefile.target.orig	2015-03-16 12:51:28.222386599 +0100
+++ netsurf-3.3/monkey/Makefile.target	2015-03-16 12:52:53.091926808 +0100
@@ -33,8 +33,6 @@ $(eval $(call pkg_config_find_and_add_en
 # no pkg-config for this library
 $(eval $(call feature_enabled,WEBP,-DWITH_WEBP,-lwebp -lvpx,WebP (libwebp)))
 
-CWARNFLAGS += -Werror
-
 CFLAGS += -std=c99 -Dmonkey -Dnsmonkey -g \
 	  -D_BSD_SOURCE \
 	  -D_XOPEN_SOURCE=600 \
