--- netsurf-full-3.0/src/netsurf-3.0/framebuffer/fbtk/event.c.ff	2013-05-03 13:30:14.068056595 +0200
+++ netsurf-full-3.0/src/netsurf-3.0/framebuffer/fbtk/event.c	2013-05-03 13:32:35.986917982 +0200
@@ -218,6 +218,12 @@ fbtk_event(fbtk_widget_t *root, nsfb_eve
 		switch (event->type) {
 		case NSFB_EVENT_KEY_DOWN:
 		case NSFB_EVENT_KEY_UP:
+			if (event->value.keycode == NSFB_KEY_F10) {
+				event->type = NSFB_EVENT_CONTROL;
+				event->value.controlcode = NSFB_CONTROL_QUIT;
+				unused = true;
+				break;
+			}
 			if ((event->value.keycode >= NSFB_KEY_MOUSE_1) &&
 			    (event->value.keycode <= NSFB_KEY_MOUSE_5)) {
 				fbtk_click(root, event);
